CDCgov/prime-reportstream
# 12153 - Added File download on Message Details

This PR ...

- Added ability to download report file within the Message Receivers table.
- Made ReportLink.tsx agnostic so that it doesn't rely on a report object.
- Added logic in the HistoryFunctions.kt to bypass org check if signed in as an Admin so that an admin can download a file.

Test Steps:
1. Log into RS
2. goto Message Id Search
3. Search for a message
4.Notice the Files in the table are able to download

## Changes
![Screenshot 2023-12-01 at 12 21 39 PM](https://github.com/CDCgov/prime-reportstream/assets/102491809/db2be5c2-9f6e-4ef1-9c52-8315ab7973d8)


## Checklist

### Testing

## Linked Issues
- Fixes #12153
